tests/cbmc/proofs/s2n_stuffer_alloc/s2n_stuffer_alloc_harness.c

```diff
@@ -14,16 +14,30 @@
  */
 
 #include "api/s2n.h"
+#include "error/s2n_errno.h"
 #include "stuffer/s2n_stuffer.h"
+#include "utils/s2n_mem.h"
+
 #include <assert.h>
 #include <cbmc_proof/proof_allocators.h>
 #include <cbmc_proof/make_common_datastructures.h>
 
 void s2n_stuffer_alloc_harness() {
-  struct s2n_stuffer *stuffer = can_fail_malloc(sizeof(*stuffer));
-  struct s2n_blob* in = cbmc_allocate_s2n_blob();
-  if (s2n_stuffer_init(stuffer, in) == S2N_SUCCESS){
-    assert(s2n_stuffer_is_valid(stuffer));
-    assert(s2n_blob_is_valid(in));
-  };
+    /* Non-deterministic inputs. */
+    struct s2n_stuffer *stuffer = nondet_bool() ? cbmc_allocate_s2n_stuffer(): NULL;
+    __CPROVER_assume(S2N_IMPLIES(stuffer != NULL, s2n_stuffer_is_valid(stuffer)));
+    uint32_t size;
+
+    /* Non-deterministically set initialized (in s2n_mem) to true. */
+    if(nondet_bool()) {
+        s2n_mem_init();
+    }
+
+    /* Operation under verification. */
+    if (s2n_stuffer_alloc(stuffer, size) == S2N_SUCCESS) {
```

Comment:
Are there any useful assertions we can make on failure?

Reply:
Yes, it does. In case of failure, we cannot guarantee the validity of the internal `s2n_blob` (e.g., in case `s2n_free` fails during `s2n_realloc`), but we can guarantee the remaining fields will be unchanged.